test_app.py

```diff
@@ -0,0 +1,21 @@
+import os
+from threading import Thread
+import app
+import pytest
+from trello_items import create_trello_board, delete_trello_board
+
+@pytest.fixture(scope='module')
+def test_app():
+    # Create the new board & update the board id environment variable
+    board_id = create_trello_board()
+    os.environ['TRELLO_BOARD_ID'] = board_id
```

Comment:
This env var TRELLO_BOARD_ID doesn't seem to get used anywhere. You just use the list ids directly. So your test is using the real lists.

Reply:
I refactored the code so that it fetches the list IDs from the environment variables each time. Then I create a temporary board and temporary lists, updating the environment variables with the IDs of the new lists for the sake of the tests.